analyses/malawi/scripts/mwi_chirps_dry_spell_viz.R

```diff
@@ -13,118 +14,174 @@ source("scripts/mwi_chirps_dry_spell_viz_hm_functions.R")
-# ggsave(output_path_hm,plot = hm_adm1, width=20,height=15)
-
-#TODO: create plots for the less than xmm per day methodology
-#ggsave(paste0(dry_spell_dir, '/dry_spell_plots/mean_back_dry_spell_hm.png'))
-#ggsave(paste0(dry_spell_dir, '/dry_spell_plots/dry_spell_hm_consecutive_days_4mm.png'), width = 7.55, height = 7.82, units = "in", dpi = 300)
-#ggsave(paste0(dry_spell_dir, '/dry_spell_plots/dry_spell_hm_consecutive_days_2mm.png'), width = 7.55, height = 7.82, units = "in", dpi = 300)
-
-
-
-
-
-### Make heatmap of the match of observed dry spells with different aggregation methodologies and different indicators
-## Mean vs >=50% pixels
-#Set general variables for heatmap
-dry_spell_match_values=c(10, 0, 13,12,11)
-match_values_labels=c("Rainy season", "Dry season", glue('mean and >={threshold} dry spell'),'mean dry spell',glue('>={threshold} dry spell'))
-color_scale=c('#b3e7ff', '#fff2d6','#0063B3' ,'#F2645A',"#78D9D1")
-y_label="Admin 2 region"
+ ## mean
+ df_dry_spells <- read.csv(paste0(dry_spell_dir,'dry_spells_during_rainy_season_list_2000_2020_mean_back.csv'))
+ df_rainy_season <- read.csv(paste0(dry_spell_dir, "rainy_seasons_detail_2000_2020_mean_back.csv"))
+ plot_title=glue("20 years of dry spells in Malawi's admin 2 regions, 2000-2020") #not indicating aggregation method as this is our standard
+ output_path_hm=paste0(exploration_dry_spell_dir, glue('mwi_viz_hm_dry_spell_mean_adm2.png'))
+ hm_mean <- plot_heatmap(df_dry_spells,df_rainy_season, dry_spell_match_values,match_values_labels,color_scale,y_label,plot_title,output_path_hm)
+ # ggsave(output_path_hm,plot = hm_mean, width=20,height=15)
 
-threshold=50
-df_dry_spells <- read.csv(paste0(exploration_dry_spell_dir,glue('dryspells_mean_pixel_th{threshold}_viz.csv')))
-df_rainy_season <- read.csv(paste0(dry_spell_dir, "rainy_seasons_detail_2000_2020_mean_back.csv"))
-plot_title=glue('Observed dry spells per admin 2 with mean and >={threshold}% aggregation methodology')
-output_path_hm=paste0(exploration_dry_spell_dir, glue('mwi_viz_hm_dry_spell_mean_th{threshold}_adm2.png'))
-hm_mean_pixel <- plot_heatmap(df_dry_spells,df_rainy_season, dry_spell_match_values,match_values_labels,color_scale,y_label,plot_title,output_path_hm,ds_flatdata=TRUE)
-# ggsave(output_path_hm,plot = hm_mean_pixel, width=20,height=15)
+ ## max
+ df_dry_spells <- read.csv(paste0(dry_spell_dir,'dry_spells_during_rainy_season_list_2000_2020.csv'))
+ df_rainy_season <- read.csv(paste0(dry_spell_dir, "rainy_seasons_detail_2000_2020.csv"))
+ plot_title=glue("20 years of dry spells in Malawi's admin 2 regions, 2000-2020")
+ sub_title = "Dry spell determined by the cell with maximum precipitation per admin2"
+ output_path_hm=paste0(exploration_dry_spell_dir, glue('mwi_viz_hm_dry_spell_max_adm2.png'))
+ hm_max <- plot_heatmap(df_dry_spells,df_rainy_season, dry_spell_match_values,match_values_labels,color_scale,y_label,plot_title,sub_title=sub_title)
+ # ggsave(output_path_hm,plot = hm_max, width=20,height=15)
```

Comment:
I've done a lot of commenting/uncommenting of plot saving lines and have just started switching to having a global ```SAVE_STATUS``` var at the top of a script to turn on/off all plot saving. This doesn't really work if you want to save some plots and not others, but can maybe help keep things a bit cleaner.

Reply:
Aah I don't know if that works in this case, since often running selection of plots, but really like the idea. Couldn't find a script where you implemented this though, would you have an example?